### PR TITLE
Range parser with real data tests

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

Parser for PgSql [range types](https://postgrespro.ru/docs/postgrespro/13/rangetypes) with real data tests

### CODE REVIEW

File: tests/RangeParserTest.php [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+            'bigint_range' => [null, '2147483650'],
+            'num_range' => [null, 20.7],
+            'ts_range' => [null, '2018-10-20 15:10:00'],
+            'ts_tz_range' => [null, '2019-10-20 15:10:00+00:00'],
+            'date_range' => [null, '2021-01-01'],
+        ],
+    ];
+
+    public function testSimpleRanges(): void
+    {
```
Comment (darkdef) [inline tests/RangeParserTest.php:43]: It's very complex tests. Please - use dataProvider and split to many small and readly tests

**Review by darkdef — CHANGES_REQUESTED** [review]

File: tests/RangeParserTest.php [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+            'num_range' => [30.7, null],
+            'ts_range' => ['2017-10-20 10:10:00', null],
+            'ts_tz_range' => ['2018-10-20 10:10:00+00:00', null],
+            'date_range' => ['2020-12-01', null],
+        ],
+        3 => [
+            'int_range' => [null, 10],
+            'bigint_range' => [null, '2147483650'],
+            'num_range' => [null, 20.7],
+            'ts_range' => [null, '2018-10-20 15:10:00'],
```
[the remark this answers is not in the document]
Comment (Gerych1984) [reply tests/RangeParserTest.php:36]: > How about add tests with negative float and integer values? And tests with inverted borders, for example [29,1]

Постгрес не даст сохранить диапазон с перепутанными границами
